mlflow/tracing/processor/inference_table.py

```diff
@@ -129,7 +130,10 @@ def on_end(self, span: OTelReadableSpan) -> None:
                 return
 
             trace.info.execution_duration = (span.end_time - span.start_time) // 1_000_000
-            trace.info.state = TraceState.from_otel_status(span.status)
+
+            # Update trace state from span status, but only if the user hasn't explicitly set
+            # a different trace status
```

Comment:
@B-Step62 great catch! I added logic to cover the inference table case, as well as tests